xtask/src/release.rs

```diff
@@ -153,9 +201,47 @@ impl ReleaseTask {
         Ok(remote)
     }
 
-    /// Get the tag name for this release.
-    fn tag_name(&self) -> String {
-        format!("{}-{}", self.local_crate.name, self.local_crate.version)
+    /// Commit and push all the changes in the git repository.
+    fn commit(&self) -> Result<()> {
+        let mut input = String::new();
+        let stdin = stdin();
+
+        print!("Ready to commit the changes. [continue/abort/DIFF]: ");
+        stdout().flush()?;
+
+        let mut handle = stdin.lock();
+        handle.read_line(&mut input)?;
+
+        match input.trim().to_ascii_lowercase().as_str() {
+            "c" | "continue" => {}
+            "a" | "abort" => {
+                return Err("User aborted commit".into());
+            }
+            _ => {
+                cmd!("git diff").run()?;
+                print!("Commit the changes? [continue/ABORT]: ");
+                stdout().flush()?;
+
+                handle.read_line(&mut input)?;
+
+                match input.trim().to_ascii_lowercase().as_str() {
+                    "c" | "continue" => {}
+                    _ => {
+                        return Err("User aborted commit".into());
+                    }
+                }
+            }
+        }
+
+        let message = format!("Release {}", self.title());
+
+        println!("Creating commit…");
+        cmd!("git commit -S -a -m {message}").read()?;
```

Comment:
I don't think signing commits should be forced One can always set `commit.gpgSign` through `git config` if that should be the default.